src/config.js

```diff
@@ -0,0 +1,1602 @@
+  }
+
+  loadUserConfig () {
+    let error
+    if (this.shouldNotAccessFileSystem()) { return }
+    if (this.savePending) { return }
+
+    try {
+      if (!fs.existsSync(this.configFilePath)) {
+        fs.makeTreeSync(path.dirname(this.configFilePath))
+        CSON.writeFileSync(this.configFilePath, {}, {flag: 'wx'}) // fails if file exists
+      }
+    } catch (error1) {
+      error = error1
+      if (error.code !== 'EEXIST') {
+        this.configFileHasErrors = true
+        this.notifyFailure(`Failed to initialize \`${path.basename(this.configFilePath)}\``, error.stack)
+        return
+      }
+    }
+
+    try {
+      let userConfig = CSON.readFileSync(this.configFilePath)
+      if (userConfig === null) { userConfig = {} }
+
+      if (!isPlainObject(userConfig)) {
+        throw new Error(`\`${path.basename(this.configFilePath)}\` must contain valid JSON or CSON`)
+      }
+
+      this.resetUserSettings(userConfig)
+      this.configFileHasErrors = false
+    } catch (error2) {
+      error = error2
+      this.configFileHasErrors = true
+      const message = `Failed to load \`${path.basename(this.configFilePath)}\``
+
+      // stack is the output from CSON in this case
+      // message will be EACCES permission denied, et al
+
+      const detail = (error.location != null) ? error.stack : error.message
+      return this.notifyFailure(message, detail)
+    }
+  }
+
+  observeUserConfig () {
+    if (this.shouldNotAccessFileSystem()) { return }
+
+    try {
+      if (this.watchSubscriptionPromise == null) {
+        this.watchSubscriptionPromise = watchPath(this.configFilePath, {}, events => {
+          return (() => {
+            const result = []
+            for (let {action} of Array.from(events)) {
+              if (['created', 'modified', 'renamed'].includes(action) && (this.watchSubscriptionPromise != null)) {
+                result.push(this.requestLoad())
+              } else {
+                result.push(undefined)
+              }
+            }
+            return result
+          })()
+        })
+      }
+    } catch (error) {
+      this.notifyFailure(`\
+Unable to watch path: \`${path.basename(this.configFilePath)}\`. Make sure you have permissions to
+\`${this.configFilePath}\`. On linux there are currently problems with watch
+sizes. See [this document][watches] for more info.
+[watches]:https://github.com/atom/atom/blob/master/docs/build-instructions/linux.md#typeerror-unable-to-watch-path\
+`
+        )
+    }
+
+    return this.watchSubscriptionPromise
+  }
+
+  unobserveUserConfig () {
+    if (this.watchSubscriptionPromise != null) {
+      this.watchSubscriptionPromise.then(watcher => watcher != null ? watcher.dispose() : undefined)
+    }
+    this.watchSubscriptionPromise = null
+  }
+
+  notifyFailure (errorMessage, detail) {
+    return (this.notificationManager != null ? this.notificationManager.addError(errorMessage, {detail, dismissable: true}) : undefined)
```

Comment:
i think this would be clearer like this:

```js
if (this.notificationManager == null) return
return this.notificationManager.addError(errorMessage, {detail, dismissable: true})
```

and honestly i doubt that the value even has to be returned.